types/schematypeoptions.d.ts

```diff
@@ -0,0 +1,219 @@
+declare module 'mongoose' {
+
+  interface ValidatorProps<T> {
+    path: string;
+    value: T;
+  }
+
+  interface ValidatorMessageFn<T> {
+    (props: ValidatorProps<T>): string;
+  }
+
+  interface ValidateFn<T> {
+    (value: T): boolean;
+  }
+
+  interface LegacyAsyncValidateFn<T> {
+    (value: T, callback: (result: boolean) => void): void;
+  }
+
+  interface AsyncValidateFn<T> {
+    (value: T): Promise<boolean>;
+  }
+
+  interface ValidateOpts<T> {
+    msg?: string;
+    message?: string | ValidatorMessageFn<T>;
+    type?: string;
+    validator: ValidateFn<T> | LegacyAsyncValidateFn<T> | AsyncValidateFn<T>;
+  }
+
+  type SchemaValidator<T> = RegExp | [RegExp, string] | Function | [Function, string] | ValidateOpts<T> | ValidateOpts<T>[];
+
+  export class SchemaTypeOptionsClass implements
```

Comment:
This seems a little strange. Why not just use `typeof SchemaTypeOptions` like we currently do?